core/chains/evm/gas/block_history_estimator.go

```diff
@@ -128,11 +127,11 @@ func NewBlockHistoryEstimator(lggr logger.Logger, ethClient evmclient.Client, cf
 
 // OnNewLongestChain recalculates and sets global gas price if a sampled new head comes
 // in and we are not currently fetching
-func (b *BlockHistoryEstimator) OnNewLongestChain(_ context.Context, head *evmtypes.Head) {
+func (b *BlockHistoryEstimator) OnNewLongestChain(_ context.Context, head txmgrtypes.Head) {
 	// set latest base fee here to avoid potential lag introduced by block delay
 	// it is really important that base fee be as up-to-date as possible
-	b.setLatest(head)
-	b.mb.Deliver(head)
+	b.setLatest(head.(*evmtypes.Head))
+	b.mb.Deliver(head.(*evmtypes.Head))
```

Comment:
not the prettiest but functionally should be fine...small suggestion could be to make the conversion just once and pass the result to the two functions rather than casting twice?

Reply:
thanks, done